modules/tensor_mechanics/include/materials/DiscreteRRIsotropicPlasticity.h

```diff
@@ -0,0 +1,62 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef DISCRETERRISOTROPICPLASTICITY_H
+#define DISCRETERRISOTROPICPLASTICITY_H
+
+#include "DiscreteRadialReturnStressIncrement.h"
+#include "PiecewiseLinear.h"
+#include "MooseMesh.h"
+
+// forward declarations
+class PiecewiseLinear;
+
+/**
+ */
+
+class DiscreteRRIsotropicPlasticity : public DiscreteRadialReturnStressIncrement
+{
+public:
+  DiscreteRRIsotropicPlasticity( const InputParameters & parameters);
+
+protected:
+  void resetQpProperties(); // Values here need to be set to a constant, ideally zero
```

Comment:
please put comment in the line above the declaration and use `///` doxygen syntax